File: python/tp1.py
class Bioskop:
    def __init__(self, IdBioskop, popcorn, softdrink, waktuBuka):
        self.__idBioskop = IdBioskop
        self.__popcorn = popcorn
        self.__softdrink = softdrink
        self.__waktubuka = waktuBuka

    # getter
    def getIdBioskop(self):
        return self.__idBioskop
    def getPopcorn(self):
        return self.__popcorn
    def getSoftdrink(self):
        return self.__softdrink
    def getWaktuBuka(self):
        return self.__waktubuka

    # setter
    def setIdBioskop(self, idbioskop):
        self.__idBioskop = idbioskop
    def setPopcorn(self, popcorn):
        self.__popcorn = popcorn
    def setSoftdrink(self, softdrink):
        self.__softdrink = softdrink
    def setWaktubuka(self, waktubuka):
        self.__waktubuka = waktubuka

# list daftar objek bioskop
daftarBioskop = []

# buat 3 object bioskop
bioskop1 = Bioskop("001", "Popcorn Balado", "Milo", "8.00")
daftarBioskop.append(bioskop1)

bioskop2 = Bioskop("002", "Popcorn Caramel", "Coca-Cola", "12.00")
daftarBioskop.append(bioskop2)

bioskop3 = Bioskop("003", "Popcorn Original", "Fanta", "9.00")
daftarBioskop.append(bioskop3)

# tambah data bioskop
print("**** TAMBAH DATA BIOSKOP ****")
idBaru = input("Masukkan ID Bioskop: ")
popcornBaru = input("Masukkan Rasa Popcorn: ")
softdrinkBaru = input("Masukkan Jenis Soft Drink: ")
waktuBukaBaru = input("Masukkan Waktu Buka: ")
bioskopBaru = Bioskop(idBaru, popcornBaru, softdrinkBaru, waktuBukaBaru)
daftarBioskop.append(bioskopBaru)
print("Data berhasil ditambahkan")
print()

# cari bioskop dengan id
cariBioskop = input("Masukkan ID Bioskop yang dicari: ")
print("**** CARI BIOSKOP ****")
dataDitemukan = False
for bioskop in daftarBioskop:
    if bioskop.getIdBioskop() == cariBioskop:
        print("Data ketemu!")
        print(f"Id Bioskop: {bioskop.getIdBioskop()} Rasa Popcorn: {bioskop.getPopcorn()} Jenis Soft Drink: {bioskop.getSoftdrink()} Jam Buka: {bioskop.getWaktuBuka()}")
        dataDitemukan = True
        break
if dataDitemukan == False:
    print("Data tidak ditemukan")
print()

# tampilan bioskop sebelum ada hapus data
print("**** DAFTAR BIOSKOP ****")
for bioskop in daftarBioskop:
    print(f"Id Bioskop: {bioskop.getIdBioskop()} Rasa Popcorn: {bioskop.getPopcorn()} Jenis Soft Drink: {bioskop.getSoftdrink()} Jam Buka: {bioskop.getWaktuBuka()}")
    print()

# hapus bioskop dengan id bioskop
idHapus = input("Masukkan ID Bioskop yang ingin dihapus: ")
print("**** HAPUS BIOSKOP ****")
dataDitemukan = False
for bioskop in daftarBioskop:
    if bioskop.getIdBioskop() == idHapus:
        daftarBioskop.remove(bioskop)
        print("Data berhasil dihapus")
        dataDitemukan = True
        break
if dataDitemukan == False:
    print("Data tidak ditemukan")
print()

# daftar bioskop setelah ada data yang dihapus
print("**** DAFTAR BIOSKOP SETELAH ADA YANG DIHAPUS ****")
for bioskop in daftarBioskop:
    print(f"Id Bioskop: {bioskop.getIdBioskop()} Rasa Popcorn: {bioskop.getPopcorn()} Jenis Soft Drink: {bioskop.getSoftdrink()} Jam Buka: {bioskop.getWaktuBuka()}")
    print()

# ubah data
idCari = input("Masukkan ID Bioskop yang ingin diubah: ")
print("**** UPDATE DATA BIOSKOP ****")
dataDitemukan = False
for bioskop in daftarBioskop:
    if bioskop.getIdBioskop() == idCari:
        popcornBaru = input("Masukkan Rasa Popcorn Baru: ")
        softdrinkBaru = input("Masukkan Jenis Soft Drink Baru: ")
        waktuBukaBaru = input("Masukkan Waktu Buka Baru: ")
        bioskop.setPopcorn(popcornBaru)
        bioskop.setSoftdrink(softdrinkBaru)
        bioskop.setWaktubuka(waktuBukaBaru)
        print("Data berhasil diupdate")
        dataDitemukan = True
        break
if dataDitemukan == False:
    print("Data tidak ditemukan")

File: java/tp1.java
import java.util.ArrayList;
import java.util.Scanner;

class Bioskop {

    private String idBioskop;
    private String popcorn;
    private String softdrink;
    private String waktuBuka;

    // constructor
    public Bioskop(String IdBioskop, String popcorn, String softdrink, String waktuBuka) {
        this.idBioskop = IdBioskop;
        this.popcorn = popcorn;
        this.softdrink = softdrink;
        this.waktuBuka = waktuBuka;
    }

    // getter
    public String getIdBioskop() {
        return idBioskop;
    }

    public String getPopcorn() {
        return popcorn;
    }

    public String getSoftdrink() {
        return softdrink;
    }

    public String getWaktuBuka() {
        return waktuBuka;
    }

    // setter
    public void setIdBioskop(String idbioskop) {
        this.idBioskop = idbioskop;
    }

    public void setPopcorn(String popcorn) {
        this.popcorn = popcorn;
    }

    public void setSoftdrink(String softdrink) {
        this.softdrink = softdrink;
    }

    public void setWaktubuka(String waktubuka) {
        this.waktuBuka = waktubuka;
    }
}


public class tp1 {

    public static void main(String[] args) {

        Scanner input = new Scanner(System.in);

        // list daftar objek bioskop
        ArrayList<Bioskop> daftarBioskop = new ArrayList<>();

        // buat 3 object bioskop
        Bioskop bioskop1 = new Bioskop(
            "001", "Popcorn Balado", "Milo", "8.00"
        );
        daftarBioskop.add(bioskop1);

        Bioskop bioskop2 = new Bioskop(
            "002", "Popcorn Caramel", "Coca-Cola", "12.00"
        );
        daftarBioskop.add(bioskop2);

        Bioskop bioskop3 = new Bioskop(
            "003", "Popcorn Original", "Fanta", "9.00"
        );
        daftarBioskop.add(bioskop3);


        // tambah data bioskop
        System.out.println("**** TAMBAH DATA BIOSKOP ****");

        System.out.print("Masukkan ID Bioskop: ");
        String idBaru = input.nextLine();

        System.out.print("Masukkan Rasa Popcorn: ");
        String popcornBaru = input.nextLine();

        System.out.print("Masukkan Jenis Soft Drink: ");
        String softdrinkBaru = input.nextLine();

        System.out.print("Masukkan Waktu Buka: ");
        String waktuBukaBaru = input.nextLine();

        Bioskop bioskopBaru = new Bioskop(
            idBaru,
            popcornBaru,
            softdrinkBaru,
            waktuBukaBaru
        );

        daftarBioskop.add(bioskopBaru);

        System.out.println("Data berhasil ditambahkan");
        System.out.println();


        // cari bioskop dengan id
        System.out.print("Masukkan ID Bioskop yang dicari: ");
        String cariBioskop = input.nextLine();

        System.out.println("**** CARI BIOSKOP ****");

        boolean dataDitemukan = false;

        for (Bioskop bioskop : daftarBioskop) {

            if (bioskop.getIdBioskop().equals(cariBioskop)) {

                System.out.println("Data ketemu!");

                System.out.println(
                    "Id Bioskop: " + bioskop.getIdBioskop()
                    + " Rasa Popcorn: " + bioskop.getPopcorn()
                    + " Jenis Soft Drink: " + bioskop.getSoftdrink()
                    + " Jam Buka: " + bioskop.getWaktuBuka()
                );

                dataDitemukan = true;

                break;
            }
        }

        if (dataDitemukan == false) {
            System.out.println("Data tidak ditemukan");
        }

        System.out.println();


        // tampilan bioskop sebelum ada hapus data
        System.out.println("**** DAFTAR BIOSKOP ****");

        for (Bioskop bioskop : daftarBioskop) {

            System.out.println(
                "Id Bioskop: " + bioskop.getIdBioskop()
                + " Rasa Popcorn: " + bioskop.getPopcorn()
                + " Jenis Soft Drink: " + bioskop.getSoftdrink()
                + " Jam Buka: " + bioskop.getWaktuBuka()
            );

            System.out.println();
        }


        // hapus bioskop dengan id bioskop
        System.out.print("Masukkan ID Bioskop yang ingin dihapus: ");
        String idHapus = input.nextLine();

        System.out.println("**** HAPUS BIOSKOP ****");

        dataDitemukan = false;

        for (int i = 0; i < daftarBioskop.size(); i++) {

            if (daftarBioskop.get(i).getIdBioskop().equals(idHapus)) {

                daftarBioskop.remove(i);

                System.out.println("Data berhasil dihapus");

                dataDitemukan = true;

                break;
            }
        }

        if (dataDitemukan == false) {
            System.out.println("Data tidak ditemukan");
        }

        System.out.println();


        // daftar bioskop setelah ada data yang dihapus
        System.out.println(
            "**** DAFTAR BIOSKOP SETELAH ADA YANG DIHAPUS ****"
        );

        for (Bioskop bioskop : daftarBioskop) {

            System.out.println(
                "Id Bioskop: " + bioskop.getIdBioskop()
                + " Rasa Popcorn: " + bioskop.getPopcorn()
                + " Jenis Soft Drink: " + bioskop.getSoftdrink()
                + " Jam Buka: " + bioskop.getWaktuBuka()
            );

            System.out.println();
        }


        // ubah data
        System.out.print("Masukkan ID Bioskop yang ingin diubah: ");
        String idCari = input.nextLine();

        System.out.println("**** UPDATE DATA BIOSKOP ****");

        dataDitemukan = false;

        for (Bioskop bioskop : daftarBioskop) {

            if (bioskop.getIdBioskop().equals(idCari)) {

                System.out.print("Masukkan Rasa Popcorn Baru: ");
                popcornBaru = input.nextLine();

                System.out.print("Masukkan Jenis Soft Drink Baru: ");
                softdrinkBaru = input.nextLine();

                System.out.print("Masukkan Waktu Buka Baru: ");
                waktuBukaBaru = input.nextLine();

                bioskop.setPopcorn(popcornBaru);
                bioskop.setSoftdrink(softdrinkBaru);
                bioskop.setWaktubuka(waktuBukaBaru);

                System.out.println("Data berhasil diupdate");

                dataDitemukan = true;

                break;
            }
        }

        if (dataDitemukan == false) {
            System.out.println("Data tidak ditemukan");
        }

        input.close();
    }
}